src/features/courses/course-controller.ts

```diff
@@ -2399,16 +2419,18 @@ class CourseController {
             const questions = await CourseWWTopicQuestion.findAll(findOptions);
             if (!_.isNil(courseTopicContentId)){
                 const topic = await this.getTopicById({id: courseTopicContentId});
-                // TODO remove assessment hardcoding
+                // TODO remove assessment hardcoding -- userId nil-check is for TS
                 if (topic.topicTypeId === 2 && !_.isNil(userId)) {
                     await questions.asyncForEach(async (question) => {
                         if (_.isNil(question.grades)) throw new RederlyExtendedError('Impossible! Found an assessment question without a grade.');
                         const version = await courseRepository.getCurrentInstanceForQuestion({questionId: question.id, userId});
                         question.webworkQuestionPath = version?.webworkQuestionPath ?? question.webworkQuestionPath;
                         question.grades.randomSeed = version?.randomSeed ?? question.grades.randomSeed;
+                        question.problemNumber = version?.problemNumber ?? question.problemNumber;
```

Comment:
Should we do this through the assign with the utility function like we have been doing elsewhere?

Reply:
agreed, but this question object is a sequelize with includes -- so it's not overwriting one of our "fundamental" objects.

Reply:
Yeah, but that is why we do the multiple interface approach and do the assign using _.pick

Reply:
Oh right - those are already implemented, I just forgot to take advantage!